ray_beam_runner/portability/ray_fn_runner.py

```diff
@@ -198,34 +198,33 @@ def run_stages(self,
         bundle_context_manager = RayBundleContextManager(
             runner_execution_context, stage)
 
+        # check input watermark
+        stage_node = ray.get(runner_execution_context.watermark_manager.get_stage_node.remote(
+          bundle_context_manager.stage.name
+        ))
+
         assert (
-            ray.get(runner_execution_context.watermark_manager.get_stage_node.remote(
-                bundle_context_manager.stage.name
-            )).input_watermark() == timestamp.MAX_TIMESTAMP), (
-            'wrong watermark for %s. Expected %s, but got %s.' % (
-                runner_execution_context.watermark_manager.get_stage_node.remote(
-                    bundle_context_manager.stage.name),
+            stage_node.input_watermark() == timestamp.MAX_TIMESTAMP), (
+            'wrong input watermark for %s. Expected %s, but got %s.' % (
+                stage_node,
                 timestamp.MAX_TIMESTAMP,
-                runner_execution_context.watermark_manager.get_stage_node.remote(
-                    bundle_context_manager.stage.name
-                ).input_watermark()
+                stage_node.input_watermark()
             )
         )
 
         stage_results = self._run_stage(
             runner_execution_context, bundle_context_manager)
 
+        # check output watermark
+        stage_node = ray.get(runner_execution_context.watermark_manager.get_stage_node.remote(
+          bundle_context_manager.stage.name
+        ))
         assert (
-            ray.get(runner_execution_context.watermark_manager.get_stage_node.remote(
-                bundle_context_manager.stage.name
-            )).input_watermark() == timestamp.MAX_TIMESTAMP), (
-            'wrong input watermark for %s. Expected %s, but got %s.' % (
-            ray.get(runner_execution_context.watermark_manager.get_stage_node.remote(
-                bundle_context_manager.stage.name)),
+            stage_node.output_watermark() == timestamp.MAX_TIMESTAMP), (
```

Comment:
This was indeed supposed to be `output_watermark` instead of `input_watermark`, right?